code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1236,9 +1236,7 @@
 /mob/living/carbon/human/revive()
 
 	if(species && !(species.flags & NO_BLOOD))
-		var/blood_reagent = get_blood_name()
-		vessel.add_reagent(blood_reagent, max_blood-vessel.total_volume)
-		fixblood()
+		restore_blood()

```

Comment:
1239-1241 can be removed, as can `restore_blood`, as this is handled at `/mob/living/proc/rejuvenate()`